.github/workflows/integrate.yaml

```diff
@@ -0,0 +1,79 @@
+name: CI
+
+on:
+  workflow_call:
+    inputs:
+      rock:
+        description: "Path to rock directory, i.e. directory containing rockcraft.yaml"
+        required: true
+        default: '.'
+        type: string
+
+jobs:
+  build-scan-test-rocks:
+    name: Build, scan, and test rock
+    runs-on: ubuntu-22.04
+    steps:
+      # Ideally we'd use self-hosted runners, but this effort is still not stable.
+      # This action will remove unused software (dotnet, haskell, android libs, codeql,
+      # and docker images) from the GH runner, which will liberate around 60 GB of storage
+      # distributed in 40GB for root and around 20 for a mnt point.
+      - name: Maximise GH runner space
+        uses: easimon/maximize-build-space@v7
+        with:
+          root-reserve-mb: 40960
+          remove-dotnet: 'true'
+          remove-haskell: 'true'
+          remove-android: 'true'
+          remove-codeql: 'true'
+          remove-docker-images: 'true'
+
+      - name: Check out code
+        uses: actions/checkout@v3
+
+      - name: Build ROCK
+        uses: canonical/craft-actions/rockcraft-pack@main
+        id: rockcraft
+        with:
+          path: ./${{ inputs.rock }}
+
+      - name: Copy ROCK to Docker
+        id: rock_in_docker
+        run: |
+          NAME=$(yq eval ".name" rockcraft.yaml)
+          VERSION=$(yq eval ".version" rockcraft.yaml)
+          ARCH=$(yq eval ".platforms | keys" rockcraft.yaml | awk -F ' ' '{print $2}')
+          ROCK="${NAME}_${VERSION}_${ARCH}"
+          sudo skopeo --insecure-policy copy oci-archive:$ROCK.rock docker-daemon:$ROCK:$VERSION
+          echo "image=$ROCK:$VERSION" >> "$GITHUB_OUTPUT"
```

Comment:
This code feels generic, and is also reproduced in the common [build_and_publish](https://github.com/canonical/charmed-kubeflow-workflows/blob/main/.github/workflows/build_and_publish_rock.yaml#L78-L84) workflow.  Can we make a workflow that encapsulates this and put it in charmed-kubeflow-workflows](https://github.com/canonical/charmed-kubeflow-workflows)?

Reply:
We cannot use workflow in steps, only actions.

>You call a reusable workflow by using the uses keyword. Unlike when you are using actions within a workflow, you call reusable workflows directly within a job, and not from within job steps.

source: https://docs.github.com/en/actions/using-workflows/reusing-workflows#calling-a-reusable-workflow

Do you think it would make sense to move this to an action?

Reply:
Moving this to a script could also help with this issue https://github.com/canonical/bundle-kubeflow/issues/763. But the idea of downloading the script from somewhere and running it doesn't appeal that much to me.

Reply:
Sent a PR there so let's move this discussion there https://github.com/canonical/charmed-kubeflow-workflows/pull/26

Reply:
:+1: